crypto/x509/x509_trs.c

```diff
@@ -143,8 +143,10 @@ int X509_TRUST_add(int id, int flags, int (*ck) (X509_TRUST *, X509 *, int),
         trtmp = X509_TRUST_get0(idx);
 
     /* OPENSSL_free existing name if dynamic */
-    if (trtmp->flags & X509_TRUST_DYNAMIC_NAME)
+    if (trtmp->flags & X509_TRUST_DYNAMIC_NAME){
         OPENSSL_free(trtmp->name);
+        trtmp->name = NULL;
```

Comment:
yes, for hygiene/style his is good.  but since line 151 assigns a new value, this isn't necessary.